src/schema/filter_artworks.js

```diff
@@ -116,7 +121,10 @@ export const FilterArtworksType = new GraphQLObjectType({
               ),
             },
             connectionFromArraySlice(hits, args, {
-              arrayLength: aggregations.total.value,
+              arrayLength: Math.min(
+                aggregations.total.value,
+                totalPages * relayOptions.size
+              ),
               sliceStart: relayOptions.offset,
             })
```

Comment:
You may be able to `merge` in your own `hasNextPage` definition, which I think is clearer than calculating a fake 'total length', just for use in a helper method in the relay lib.

something like:

``` javascript
return merge(
  { pageCursors }, connectionFromArraySlice(...), { pageInfo: { hasNextPage: relayOptions.page < totalPages } }
) 
```

This way feels more explicit (we're handrolling our own implementation of `hasNextPage`, rather than munging params and getting the relay graphql lib to return the correct value).

We do something similar in:
https://github.com/artsy/metaphysics/blob/e882ea9a3e447d2afc2b13c40198e0f4af55207d/src/schema/artist/shows.js#L91-L103

Reply:
I think it's safer to just let the `connectionFromArraySlice` function do the math otherwise the connection data could be inconsistent. The `connectionFromArraySlice` function is also responsible for slicing and generating [the "right" `edges`](https://github.com/graphql/graphql-relay-js/blob/88a2c88f53634cc7cae7ff56e33f678b9f34b7d5/src/connection/arrayconnection.js#L123), so patching a piece of data seems a bit risky to me.

Reply:
Up to you! Thinking about it more, I still think patching is safe since it clearly overwrites the one piece of data we in fact _do_ want to overwrite. It also means we can keep passing the _same_ params we've been passing this entire time (so the connection data won't be inconsistent).

Your example winds up _adjusting_ the params we pass in to the library, which seems like it's a bit 'side-effect-y', as well as the potential for possibly new behavior, as we are now actually passing in new params to that lib. Whereas the `merge` suggestion keeps everything about that lib's use the same (so the behavior won't change), save for the explicit override of a single computed field.

I'm for sure merging this! Curious if i was able to convince you though!